scripts/es_index_comparison/README.md

```diff
@@ -0,0 +1,160 @@
+# es_index_comparison
+
+A Python CLI (uv project) to fetch documents from two Elasticsearch indices, materialize them locally (NDJSON gzip + Parquet shards), and compute deep, field-level diffs with flexible ignore patterns. Multiple analyses can be run side-by-side—each with its own YAML config and output namespace.
+
+## Features
+- Fetch only (read-only) from Elasticsearch using cloud_id + api_key (never writes back).
+- Optional filter query to restrict scanned documents.
+- Chunked Parquet materialization (polars + pyarrow) for efficient re-use.
+- Deep recursive diff (dicts, lists, scalars, numpy arrays) with null-vs-missing equivalence.
+- Ignore field patterns with wildcards (`*`, `**`), list index wildcards (`[]`), and descendant matching.
+- Export artifacts: JSONL diffs, CSV summary, field frequency JSON, and diff metadata.
+- Sampled human-readable diff display & per-document diff inspection.
+- Multiple independent analyses distinguished by namespace (auto or user-provided).
+
+## Installation (using uv)
+```bash
+# From repository root
+uv sync
+# or inside this subproject directory
+uv pip install -e .
```

Comment:
I think uv sync works from here and not repository root, as this is where pyproject.toml is